runtime/rococo/src/validator_manager.rs

```diff
@@ -16,84 +16,87 @@
 
 //! A pallet for managing validators on Rococo.
 
-use frame_support::{decl_error, decl_event, decl_module, decl_storage, traits::EnsureOrigin};
 use sp_staking::SessionIndex;
 use sp_std::vec::Vec;
 
+pub use pallet::*;
+
 type Session<T> = pallet_session::Pallet<T>;
 
-/// Configuration for the parachain proposer.
-pub trait Config: pallet_session::Config {
-	/// The overreaching event type.
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{dispatch::DispatchResult, pallet_prelude::*, traits::EnsureOrigin};
+	use frame_system::pallet_prelude::*;
 
-	/// Privileged origin that can add or remove validators.
-	type PrivilegedOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
-}
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
 
-decl_event! {
-	pub enum Event<T> where ValidatorId = <T as pallet_session::Config>::ValidatorId {
-		/// New validators were added to the set.
-		ValidatorsRegistered(Vec<ValidatorId>),
-		/// Validators were removed from the set.
-		ValidatorsDeregistered(Vec<ValidatorId>),
-	}
-}
+	/// Configuration for the parachain proposer.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_session::Config {
+		/// The overreaching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-decl_error! {
-	pub enum Error for Module<T: Config> {}
-}
+		/// Privileged origin that can add or remove validators.
+		type PrivilegedOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+	}
 
-decl_storage! {
-	trait Store for Module<T: Config> as ParachainProposer {
```

Comment:
the pallet prefix is changed from ParachainProposer to ValidatorManager, I guess we should migrate the storage

Reply:
Yeah I know, but the storage items are just for storing temporary data. Aka, we don't really need to migrate them.

Reply:
ok then all good to me